pkg/apis/nodeup/config.go

```diff
@@ -53,6 +53,8 @@ type Config struct {
 
 	// DefaultMachineType is the first-listed instance machine type, used if querying instance metadata fails.
 	DefaultMachineType *string `json:",omitempty"`
+	// EnableLifecycleHook defines whether we need to complete a lifecycle hook.
+	EnableLifecycleHook bool `json:",omitempty"`
```

Comment:
I've been leaning towards putting as much of the small stuff into the userdata as possible. If we can gain confidence that we have enough testing to catch any breakage in the AuxConfig mechanism, I'd like to have nodeup skip loading the AuxConfig in the common case where everything fits in userdata.

Reply:
Ah, I see. If you have to go to the state store, you might as well get everything from there instead of going field-by-field. So we'd put everything in `Config` and when it's too big, spill it to the state store and put a smaller bootstrap thing in userdata.

I'll put it on the backlog, but unblocking v1beta1 apiversion is higher priority.